plugin/src/main/java/org/opensearch/ml/utils/TenantAwareHelper.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.utils;
+
+import java.util.Objects;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.ml.settings.MLFeatureEnabledSetting;
+
+import lombok.Getter;
+import lombok.Setter;
+
+public class TenantAwareHelper implements TenantAware {
+
+    private final MLFeatureEnabledSetting mlFeatureEnabledSetting;
+    @Getter
+    @Setter
+    private String tenantId;
+
+    public TenantAwareHelper(MLFeatureEnabledSetting mlFeatureEnabledSetting) {
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
+    }
+
+    @Override
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    @Override
+    public boolean validateTenantId(ActionListener<?> listener) {
+        if (mlFeatureEnabledSetting.isMultiTenancyEnabled() && getTenantId() == null) {
+            listener.onFailure(new OpenSearchStatusException("You don't have permission to access this resource", RestStatus.FORBIDDEN));
+            return false;
+        } else
+            return true;
```

Comment:
Suggest always add `{}` even for one line code. https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces
```suggestion
        } else {
            return true;
        }
```